File: Java Desktop Runtime/jruntime/AnimationManager.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package jruntime;

import java.util.*;

/**
 *
 * @author Riju
 */
public class AnimationManager 
{
     static ArrayList<Animation> animation_list = new ArrayList<Animation>();

        public static void registerAnimation(Animation animation)
        {
            animation_list.add(animation);
            if (!animation.isPlaying()) animation.start();
        }

        public static void updateAnimation()
        {
            int cnt = 0;
            boolean toRepeat = false;
            
            do
            {
                toRepeat = false;
           
            for(;cnt < animation_list.size( );cnt++)
            {
                Animation anim = animation_list.get(cnt);

                if (!anim.isPlaying( ))
                {
                    animation_list.remove(cnt);
                    toRepeat = true;
                    break;
                }
                else
                {
                    anim.update();
                }
            }
            
            } while (toRepeat);

        }
}

File: C++ Runtime/AnimationManager.cpp
#include "AnimationManager.h"

static std::list<Animation_ptr> animation_list;

void AnimationManager::registerAnimation(Animation_ptr anim_ptr)
{
	animation_list.push_back(anim_ptr);
	if (!anim_ptr->isPlaying()) anim_ptr->start();
}

void AnimationManager::updateAnimation()
{
	int cnt = 0;
x:

	for (; cnt < animation_list.size( ); cnt++)
	{
		int cntr = 0;
		Animation_ptr anim_ptr = NULL;
		std::list<Animation_ptr>::iterator anim_iter;

		for (std::list<Animation_ptr>::iterator cur_anim = animation_list.begin(); cur_anim != animation_list.end(); cur_anim++,cntr++)
		{
			if (cntr == cnt)
			{
				anim_ptr = *cur_anim;
				anim_iter = cur_anim;
			}
		}


		if (!anim_ptr->isPlaying())
		{
			animation_list.erase(anim_iter);
			goto x;
		}
		else
		{
			anim_ptr->update();
		}
	}
}